src/dectris/neggia/test/Test_H5DataspaceMsg.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+MIT License
+
+Copyright (c) 2017 DECTRIS Ltd.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include <dectris/neggia/data/H5DataspaceMsg.h>
+#include <gtest/gtest.h>
+
+class TestH5DataspaceMsgV1 : public ::testing::Test {
+public:
+    void SetUp() {
+        // clang-format off
+        unsigned char buf[56] = {
+            0x01, 0x03, 0x01, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x05, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x0d, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x0b, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0xff, 0xff, 0xff, 0xff,
+            0xff, 0xff, 0xff, 0xff,
+            0x0d, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x0b, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+        };
+        // clang-format on
+        memcpy(data, buf, 56);
+        version = 1;
+        rank = 3;
+        maxDimsPresent = true;
+        dims = std::vector<uint64_t>{5, 13, 11};
+        maxDims = std::vector<uint64_t>{0xffffffffffffffff, 13, 11};
+    };
+
+    char data[56];
+    uint8_t version;
+    uint8_t rank;
+    bool maxDimsPresent;
+    std::vector<uint64_t> dims;
+    std::vector<uint64_t> maxDims;
+};
+
+TEST_F(TestH5DataspaceMsgV1, CanBeParsed) {
+    auto msg = H5DataspaceMsg((const char*)data, 0);
+    ASSERT_EQ(msg.version(), version);
+    ASSERT_EQ(msg.rank(), rank);
+    ASSERT_EQ(msg.maxDims(), maxDimsPresent);
+    for (int i = 0; i < rank; i++) {
+        EXPECT_EQ(msg.dim(i), dims.at(i));
+        EXPECT_EQ(msg.maxDim(i), maxDims.at(i));
+    }
+}
```

Comment:
Why not just use the `TEST` macro instead and move all the code into the test function? The class is only boilerplate in this case.